tree/dataframe/inc/ROOT/RDF/RResultMap.hxx

```diff
@@ -96,19 +96,43 @@ class RResultMap {
       }
    }
 
+   void RunEventLoopIfNeeded()
+   {
+      if ((fVariedAction != nullptr && !fVariedAction->HasRun()) || !fNominalAction->HasRun())
+         fLoopManager->Run();
+   }
+
 public:
+   using iterator = typename decltype(fMap)::iterator;
+   using const_iterator = typename decltype(fMap)::const_iterator;
+
    // TODO: can we use a std::string_view here without having to create a temporary std::string anyway?
    T &operator[](const std::string &key)
    {
       auto it = fMap.find(key);
       if (it == fMap.end())
          throw std::runtime_error("RResultMap: no result with key \"" + key + "\".");
 
-      if ((fVariedAction != nullptr && !fVariedAction->HasRun()) || !fNominalAction->HasRun())
-         fLoopManager->Run();
+      RunEventLoopIfNeeded();
       return *it->second;
    }
 
+   /// Iterator to walk through key-value pairs of all variations for a given object.
+   /// Runs the event loop before returning if necessary.
+   iterator begin()
+   {
+      RunEventLoopIfNeeded();
+      return fMap.begin();
+   }
+   const_iterator cbegin()
+   {
+      RunEventLoopIfNeeded();
+      return fMap.cbegin();
+   }
+   iterator end() { return fMap.end(); }
+   const_iterator end() const { return fMap.cend(); }
```

Comment:
Nobody will ever do `return *(rmap.end() - 1);`, right? :-) Otherwise you'll have to `RunEventLoopIfNeeded()` also for `end()` and siblings.

Reply:
Ouch, yes, they might 🙁

Reply:
the backing STL container is an unordered map, so I hope not...